contrib/babelfishpg_tsql/src/forjson.c

```diff
@@ -145,6 +160,64 @@ tsql_query_to_json_internal(const char *query, int mode, bool include_null_value
 		appendStringInfoString(result, "]}");
 	else if (!without_array_wrapper)
 		appendStringInfoChar(result,']');
-
 	return result;
 }
+
+/*
+ * For now report an ERROR if any attribute is binary, datetime and bit datatypes since they are not implemented yet.
+ * Also Exact numeric's datatype money or smallmoney is not supported.
+ */
+static void
+tsql_unsupported_datatype_check(void)
+{
+	for (int i = 1; i <= SPI_tuptable->tupdesc->natts; i++)
+	{
+		/* 
+		 * This part of code is a workaround for is_tsql_x_datatype() which does not work as expected, 
+		 * it compares the datatype oid of the columns with the tsql_datatype_oid and
+		 * then throw feature not supported error based on the typename.
+		 */
+		Oid nspoid;
+		Oid tsql_datatype_oid;
+
+		Oid datatype_oid = SPI_gettypeid(SPI_tuptable->tupdesc, i);
+
+		char* typename = SPI_gettype(SPI_tuptable->tupdesc, i);
+
+		nspoid = get_namespace_oid("sys", true);
+		Assert(nspoid != InvalidOid);
+
+		tsql_datatype_oid = GetSysCacheOid2(TYPENAMENSP, Anum_pg_type_oid, CStringGetDatum(typename), ObjectIdGetDatum(nspoid));
+
+		if (tsql_datatype_oid == datatype_oid)
```

Comment:
Please add a comment explaining when `tsql_datatype_oid` could be different from `datatype_oid`.